File: 36-valid-sudoku/valid-sudoku.java
class Solution {
public boolean isValidSudoku(char[][] board) {
boolean[][] rowCheck = new boolean[9][9];
boolean[][] colCheck = new boolean[9][9];
boolean[][] gridCheck = new boolean[9][9];

for (int row = 0; row < 9; row++) {
for (int col = 0; col < 9; col++) {
if (board[row][col] == '.') continue; // Skip empty cells

int num = board[row][col] - '1'; // Convert '1'-'9' to 0-8
int gridIndex = (row / 3) * 3 + (col / 3); // Identify 3x3 grid

// Check if num is already present in row, column, or grid
if (rowCheck[row][num] || colCheck[col][num] || gridCheck[gridIndex][num]) {
return false;
}

// Mark number as seen
rowCheck[row][num] = true;
colCheck[col][num] = true;
gridCheck[gridIndex][num] = true;
}
}
return true;
}
}

File: 36-valid-sudoku/valid-sudoku.cs
public class Solution {
    public bool IsValidSudoku(char[][] board) {
        int[,] RowCheck=new int[9,9];
        int[,] ColCheck=new int[9,9];
        int[,] GridCheck=new int[9,9];
        for(int i=0;i<9;i++){
            for(int j=0;j<9;j++){
                if(board[i][j]=='.') continue;
                int num=board[i][j]-'1';
                int idx=i/3*3+j/3;
                if(RowCheck[i,num]==1||ColCheck[j,num]==1||GridCheck[idx,num]==1) return false;
                RowCheck[i,num]=1;
                ColCheck[j,num]=1;
                GridCheck[idx,num]=1;
            }
        }
        return true;
    }
}